lib/js/src/manager/permission/_PermissionManagerBase.js

```diff
@@ -64,7 +65,11 @@ class _PermissionManagerBase extends _SubManagerBase {
 
         this._onPermissionsChangeListener = (notification) => {
             const permissionItems = notification.getPermissionItem();
-            const previousPermissionItems = JSON.parse(JSON.stringify(this._currentPermissionItems));
+            const previousPermissionItemsJson = JSON.parse(JSON.stringify(this._currentPermissionItems));
+            const previousPermissionItems = {};
+            Object.entries(previousPermissionItemsJson).map((functionId, permissionObject) => {
```

Comment:
Object.entries returns an array of arrays. When mapping over this, the `functionId` parameter holds the two desired values and `permissionObject` is holding the index. Instead, the signature should be: 
```
.map(([functionId, permissionObject]) => {
})

OR

.map(pair => {
   const functionId = pair[0]
   const permissionObject = pair[1]
})
```
See here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries#examples